ghcide/src/Development/IDE/Plugin/Completions.hs

```diff
@@ -209,7 +209,7 @@ extendImportHandler' ideState ExtendImport {..}
                 it = case thingParent of
                   Nothing -> newThing
                   Just p  -> p <> "(" <> newThing <> ")"
-            t <- liftMaybe $ snd <$> newImportToEdit n (astA ps)
+            t <- liftMaybe $ snd <$> newImportToEdit n (astA ps) ""
```

Comment:
Will this do the right thing if the module has a module name but no imports?

Reply:
I see what you mean, because of the empty string? I think it will be fine for that case as if the module has a name but no imports the new import will be added directly under the module declaration via the 'hsmodName' type, or via 'hsmodExports' if the declaration contains explicit exports.

But it would be an issue actually when the module has neither a module name nor imports - in that case it relies on the 'fileContents' argument when calling 'findNextPragmaPosition', but if it's just getting an empty string from there it won't always be accurate. 

I guess calling 'GetFileContents' earlier in the "extendImportHandler'" function, and then having (fromMaybe "" contents) in the line you highlighted should cover that case? I can check this evening and add a test too.